components/org.wso2.carbon.identity.oauth.endpoint/src/main/java/org/wso2/carbon/identity/oauth/endpoint/authz/OAuth2AuthzEndpoint.java

```diff
@@ -4746,4 +4768,22 @@ private Response handleUnsupportedGrantForApiBasedAuth() {
                 new AuthServiceClientException(AuthServiceConstants.ErrorMessage.ERROR_INVALID_AUTH_REQUEST.code(),
                         "App native authentication is only supported with code response type."), log);
     }
+
+    private String addServiceProviderIdToRedirectURI(String redirectURI, String serviceProviderId) {
+
+        if (StringUtils.isNotBlank(serviceProviderId)) {
+            try {
+                URI uri = new URI(redirectURI);
+                String query = uri.getRawQuery();
+                if (StringUtils.isNotBlank(query) && !query.contains(SERVICE_PROVIDER_ID + "=")) {
```

Comment:
There is a flow in this logic here. If query is not null and query contains `SERVICE_PROVIDER_ID `, this will go to the else block and add `SERVICE_PROVIDER_ID ` with a `?`.

Reply:
Addressed with https://github.com/wso2-extensions/identity-inbound-auth-oauth/pull/2588/commits/84fc98ab729c551156c0593622cdcd3f7cd52360